map.js

```diff
@@ -103,20 +101,38 @@ function drawBaseMap(){
 function initOverlay(){
     overlay = {
         locations : [],
+        tsp_locations : [],
         nirnroots : [],
+        tsp_nirnroots : [],
         lastZoomLevel : zoomLevel
     }
 
     runOnTree(jsondata.location, function(loc){
         overlay.locations.push(new MapIcon(loc));
+        //takes only what we need for TSP: x, y, and order. 
+            //should we make a seperate "TSP object?" or just leave this as is?
+        overlay.tsp_locations.push({x:loc.x, y:loc.y, tsp:loc.tsp, cell:{x:loc.x, y:loc.y}});
```

Comment:
The object for the tsp array is thrown together here. 
It's not super useful other than being organized and holding the points to draw the path from.

This could be its own more complex/formal object if wanted. but i figured it was simple enough we didn't need more than this.